### PR TITLE
Feature/sc 23053/dataset 2 0 refinement

### PR DESCRIPTION
**Pull request by popcornylu**
- Use dataset id only (no dataset name,  just like github and huggingface does)
- Close the dialog when clicking outside of the dialog
- Filter out '.dataset' in the browser
- The search should also consider the tag column
- Change the "Last modified" if files upload/delete in dataset UI.
- Refresh the "Dataset size" if files upload/delete in dataset UI.

# Screenshots
Use id as the name
image

search on tag
image

no `.dataset`
image

# Review image
- ~sc23053-c590662~
- sc23053-9f0c110

### CODE REVIEW

File: packages/client/src/components/Browser/Browser.tsx [hunk 1]
```diff
@@ -484,6 +489,18 @@ function Browser(props: BrowseInternalProps) {
   if (!isEmpty(searchText)) {
     dataSource = dataSource.filter(item => item.name.includes(searchText));
   }
```
Comment (ctiml) [inline packages/client/src/components/Browser/Browser.tsx:491]: We don't need this. It is duplicated.

**Comment by ctiml** [comment]
Since we use dataset name as an unique id, should we make it case-insensitive?

**Comment by popcornylu** [comment]
> Since we use dataset name as an unique id, should we make it case-insensitive?

Good point. But i would like to create a new story about this. Because I don't think there are apparent issue for case-sensitive implement. But i agree with u that case-insensitive would be more reasonable.

File: packages/client/src/components/Browser/Browser.tsx (hunk 1 above)
Comment (popcornylu) [reply packages/client/src/components/Browser/Browser.tsx]: fixed

**Review by ctiml — APPROVED** [review]

**Pull request merged**